### ISSUE TITLE
Custom checks not shown and no statistics

### ISSUE THREAD

**Issue by ramonsmits** [Particular/ServicePulse#349]
I created a custom check but it is not listed in the custom checks:

![image](https://cloud.githubusercontent.com/assets/152998/15394076/f4364f50-1dd0-11e6-81d1-77bfbc3e3dee.png)

I expected *all* custom checks to be visible. Not only failed checks.

I expected to see statistics too like when was the last time the custom check send a result to service control. What was the last time it failed or maybe even all failure periods.

**Comment by WojcikMike** [comment]
ATM the custom checks screen display only failed checks which I find confusing. I think ti should be changed to include all ckecks with an indication if the check is success or failure and probably more detaild indication what was the last reported time. 

@Particular/servicepulse-maintainers please have a look

**Comment by WilliamBZA** [comment]
I propose the following tasks for this as a starting point:

* [ ] Change the UI to a simple table listing the failed checks with an icon to indicate their failed state and no details, grouped by category
* [ ] Adapt the failed checks list to allow clicking on a failed check to see the details of the failures (expand/collapse the current row)
* [ ] Change `getFailingCustomChecks` to get all custom checks and display the results in the table
* [ ] Display only failing checks in the first table (clientside filtering)
* [ ] Add a similar table for passing custom checks with a different icon

Following on from this the UI should probably be revisited, cleaned up and brought in line with the rest of the ServicePulse UI.

**Comment by sergioc** [comment]
>  indicate their failed state and no details

is that because we have no control over the format of the details? E.g. they can be unstructured and/or long?

> grouped by category

how are categories defined?

Do you have any idea how many custom checks do customers tend to configure?

**Comment by WilliamBZA** [comment]
> is that because we have no control over the format of the details? E.g. they can be unstructured and/or long?

Correct. There doesn't appear to be a limit anywhere.

> how are categories defined?

Each Custom Check defines it's category at compilation time

**Comment by sergioc** [comment]
>> Do you have any idea how many custom checks do customers tend to configure?
No, so paging is essential

> No, so paging is essential

How about contacting some TAMs and ask if they have any idea how are customers using custom checks?

> Each Custom Check defines it's category at compilation time

I'm not sure I understand the purpose of categories. Is there a doco for this?

**Comment by WilliamBZA** [comment]
> I'm not sure I understand the purpose of categories. Is there a doco for this?

Nothing at all. Perhaps it's time to create some?

**Comment by sergioc** [comment]
> Perhaps it's time to create some?

Probably a good idea, but it'll likely take time.

Can you summarize the concept and purpose here?

**Comment by WilliamBZA** [comment]
I don't think that's within the scope of this issue.

**Comment by sergioc** [comment]
It's important that we understand what categories are to validate the requirement for having the display "grouped by categories".

**Comment by WilliamBZA** [comment]
If you'd prefer we can remove the group by categories and simply list the custom checks. I don't really understand the concern though. Can you elaborate and perhaps we can come to a solution?

**Comment by sergioc** [comment]
The concern is that I don't understand categories here. That's why I asked for a quick summary.

**Comment by WojcikMike** [comment]
As currently categories are not used, we can define the meaning. The way how I would use it would be either: business alignement:
 - payment - grouping all checks for payment services
 - account managment 

Or as technical alignement:
 - Database connectivity 
 - Disk availability check

That would allow to display all checks together. However I would suggest creating separate issue for categories, and for now just display it on a screen.

**Comment by sergioc** [comment]
So you just assign a specific category to a specific custom check. Got it.

If the data is available, I think it's not wise to eye-drop categories to another issue. It's just better to design it all together.
